qiskit/pulse/parameter_manager.py

```diff
@@ -410,21 +411,23 @@ def _unroll_param_dict(
             A dictionary from parameter to value.
         """
         out = {}
+        param_name_dict = {param.name: param for param in self.parameters}
```

Comment:
This does not account for the fact that more than one parameter could have the same name, which pulse allows but circuit does not (it gives an error when trying to add the second parameter).

I think would be best to loop over `self.parameters` and make one dict shaped like `dict[str, list[Parameter | ParameterVector]]` mapping each name to the parameters with that name. Then make a new dict and loop over `parameter_binds` copying in the key and values for when the key is a `Parameter` or `ParameterVector` and replacing the key when it is a string with an entry for each value in the first dict for that string (could be more than one parameter). Then loop over that dict like below but without accounting for string keys.

You may want to account for a name being used for a parameter and a parameter vector. There is validation for vectors but not for scalars. If you tried to assign to a list a string name used by both a `ParameterVector` and a `Parameter`, it would not catch the assignment of a list to a `Parameter` here.

Maybe it is worth testing duplicated names?

Reply:
Every `Parameter` carrying the same name should be assigned the same value when a string is assigned a value. Also, every `ParameterVector` should be assigned the same list of values.